packages/core/src/components/renvooi/renvooi.tsx

```diff
@@ -1,31 +1,78 @@
-import { Component, ComponentInterface, Fragment, FunctionalComponent, Prop, h } from "@stencil/core";
+import {
+  Component,
+  ComponentInterface,
+  Fragment,
+  FunctionalComponent,
+  Prop,
+  h,
+  Event,
+  EventEmitter,
+} from "@stencil/core";
 
-import { RenvooiValue } from "./renvooi.interfaces";
+import {
+  RenvooiMarkFunction,
+  RenvooiMarkItemHighlightEvent,
+  RenvooiRenderMarkFunction,
+  RenvooiValue,
+} from "./renvooi.interfaces";
 
 interface RenvooiRenderProps {
   value: RenvooiValue;
+  mark?: RenvooiRenderMarkFunction;
+  emitMarkItemHighlight(text: string, elementRef: HTMLElement): void;
 }
 
-const RenvooiRender: FunctionalComponent<RenvooiRenderProps> = ({ value }) => {
+const RenvooiRender: FunctionalComponent<RenvooiRenderProps> = ({ value, mark, emitMarkItemHighlight }) => {
   if (typeof value === "string" || !value) {
     // This element is used for --_dso-renvooi-text-decoration
-    return <span class="text">{value}</span>;
+    return <span class="text">{renderText(value)}</span>;
   }
 
   if ("toegevoegd" in value) {
-    return <ins>{value.toegevoegd}</ins>;
+    return <ins>{renderText(value.toegevoegd)}</ins>;
   }
 
   if ("verwijderd" in value) {
-    return <del>{value.verwijderd}</del>;
+    return <del>{renderText(value.verwijderd)}</del>;
   }
 
   return (
     <>
-      <del>{value.was}</del>
-      <ins>{value.wordt}</ins>
+      <del>{renderText(value.was)}</del>
```

Comment:
Je kunt hier ook de fragment wrapper weg laten. JSX kan meerdere elementen in een array renderen. Kijk maar wat je mooier vindt.

```suggestion
  return [<del>{renderText(value.was)}</del>, <ins>{renderText(value.wordt)}</ins>];
```

Reply:
Heb je suggestie overgenomen.